x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -45,4 +56,51 @@ public LogicalPlan optimize(LogicalPlan verified) {
 
         return Arrays.asList(operators, label);
     }
+
+
+    private static class ReplaceWildcards extends OptimizerRule<Filter> {
+
+        private static boolean isWildcard(Expression expr) {
+            if (expr.foldable()) {
+                Object value = expr.fold();
+                return value instanceof String && value.toString().contains("*");
+            }
+            return false;
+        }
+
+        private static LikePattern toLikePattern(String s) {
+            // pick a character that is guaranteed not to be in the string, because it isn't allowed to escape itself
+            char escape = 1;
```

Comment:
Why not 0?

Reply:
Looks like there's explicit handling for `\0`
https://github.com/elastic/elasticsearch/blob/bf317e8c4eb301f5ea9ff17a3d9595ae177cfbaf/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/StringUtils.java#L104